data/src/main/java/edu/nextstep/camp/calculator/data/CalculatorRepositoryImpl.kt

```diff
@@ -0,0 +1,38 @@
+package edu.nextstep.camp.calculator.data
+
+import edu.nextstep.camp.calculator.domain.CalculationRecord
+import edu.nextstep.camp.calculator.domain.CalculatorRepository
+
+/**
+ * 클래스에 대한 간단한 설명이나 참고 url을 남겨주세요.
+ * Created by jeongjinhong on 2022. 07. 31..
+ */
+
+internal class CalculatorRepositoryImpl(private val calculationRecordsDao: CalculationRecordDao) :
+    CalculatorRepository {
+
+    private var calculationRecordList: MutableList<CalculationRecord> = mutableListOf()
+
+    private suspend fun getCalculationRecords(): List<CalculationRecord> {
+        return calculationRecordsDao.getAll().map { it.mapToDomain() }
+    }
+
+    private suspend fun insert(calculationRecord: CalculationRecord) {
+        return calculationRecordsDao.insert(CalculationRecordEntity.of(calculationRecord))
+    }
+
+    override suspend fun storeCalculationMemory(calculationRecord: CalculationRecord) {
+        calculationRecordList.add(calculationRecord)
+
+        insert(calculationRecord)
+    }
+
+    override suspend fun loadCalculationRecords() {
+        calculationRecordList = getCalculationRecords().toMutableList()
+    }
+
+    override fun getCalculationRecordList(): MutableList<CalculationRecord> {
+        return calculationRecordList
+    }
+
+}
```

Comment:
```suggestion
internal class CalculatorRepositoryImpl(private val calculationRecordsDao: CalculationRecordDao) :
    CalculatorRepository {

    private suspend fun getCalculationRecords(): List<CalculationRecord> {
        return calculationRecordsDao.getAll().map { it.mapToDomain() }
    }

    private suspend fun insert(calculationRecord: CalculationRecord) {
        return calculationRecordsDao.insert(CalculationRecordEntity.of(calculationRecord))
    }

    override suspend fun storeCalculationMemory(calculationRecord: CalculationRecord) {
        insert(calculationRecord)
    }

    override suspend fun loadCalculationRecords(): List<CalculationRecord> {
        return getCalculationRecords().toMutableList()
    }
}
```

제가 원한 형태는 위와 같습니다. calculationRecordList 역할이 어떤건지 여전히 모호합니다. 🤔
왜 중간에 값을 저장하고 있다가 이 값을 전달해야 되는지 의문이긴 합니다.
캐시 역할을 하고 있는것처럼 보이기도 합니다만 실제 캐시로 구현을 하신건지도 궁금하네요.

상태 저장, 변경없이 프로그래밍 한다는 것은 불가능 합니다. 다만 그런 역할을 하는 함수를 제외하고는
순수함수 형태로 작성할 수 있는 부분들은 최대한 그렇게 작성해 주시면 좋기 때문에 코멘트 해드렸습니다.

고민해보셨으면 해서 간단한 링크 남겨드립니다. https://velog.io/@chdb57/%E3%85%87%E3%85%87%E3%85%87%E3%85%87%E3%85%87%E3%85%87%E3%85%87%E3%85%87